### PR TITLE
Api - Reference Data Type Id

### PR DESCRIPTION
**Pull request by RafaelMeha**
Created get method for reference data by type and id

### CODE REVIEW

File: RealestateApi/Controllers/ReferenceDataController.cs [hunk 1]
```diff
@@ -44,5 +44,11 @@ public async Task<ReferenceDataResponseDto> DeleteReferenceDataAsync(string refD
                 throw;
             }
         }
+
+        [HttpGet("{refDataType}/{refDataId}", Name = "ReferenceData")]
+        public async Task<ReferenceDataModel> GetReferenceDataAsync(string refDataType, int refDataId)
```
Comment (Silva096) [inline RealestateApi/Controllers/ReferenceDataController.cs:49]: Change name from "GetReferenceDataAsync" to "GetReferenceDataByIdAsync" please

File: RealestateApi/Repository/ReferenceDataRepository.cs [hunk 2]
```diff
@@ -213,5 +213,93 @@ public async Task<ReferenceDataResponseDto> DeleteAmenityReferenceDataAsync(stri
...
+                ReferenceDataModel response = new()
+                {
+                    Id = (int)typologyReader["id"],
+                    Description = (string)typologyReader["description"],
+                };
+
+                return response;
+            }
+
+            return null;
```
Comment (Silva096) [inline RealestateApi/Repository/ReferenceDataRepository.cs:236]: You should not return null, you should return response.

By returning response inside a while loop, you will return as soon as the first one is read.

**Review by Silva096 — CHANGES_REQUESTED** [review]
Comments

File: RealestateApi/Repository/ReferenceDataRepository.cs [hunk 3]
```diff
@@ -216,90 +216,119 @@ public async Task<ReferenceDataResponseDto> DeleteAmenityReferenceDataAsync(stri
...
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine(ex.Message);
             }
 
-            return null;
+            return response;
         }
+
```
Comment (jonathan-osorio) [inline RealestateApi/Repository/ReferenceDataRepository.cs:332]: extra line not needed

**Review by jonathan-osorio — APPROVED** [review]

File: RealestateApi/Repository/ReferenceDataRepository.cs (hunk 3 above)
Comment (RafaelMeha) [reply RealestateApi/Repository/ReferenceDataRepository.cs:332]: thanks :)

**Review by Silva096 — APPROVED** [review]

**Pull request merged**